pkg/config/polling_manager_test.go

```diff
@@ -146,6 +150,127 @@ func TestNewAsyncPollingProjectConfigManagerWithNullDatafile(t *testing.T) {
 	mockRequester.AssertExpectations(t)
 }
 
+func TestGetConfigForPollingConfigManagerWaitsForTimeoutForInvalidDatafile(t *testing.T) {
+	blockingTimeout := 50 * time.Millisecond
+	mockDatafile := []byte("NOT-VALID")
+	mockRequester := new(MockRequester)
+	mockRequester.On("Get", []utils.Header(nil)).Return(mockDatafile, http.Header{}, http.StatusOK, nil)
+
+	sdkKey := "test_sdk_key"
+
+	configManager := NewPollingProjectConfigManager(sdkKey, WithRequester(mockRequester), WithBlockingTimeout(blockingTimeout))
+	mockRequester.AssertExpectations(t)
+
+	start := time.Now()
+	expectedExecutionTime := start.Add(blockingTimeout)
+	_, _ = configManager.GetConfig()
+	actualExecutionTime := start.Add(time.Since(start))
+	assert.WithinDuration(t, expectedExecutionTime, actualExecutionTime, 10*time.Millisecond)
+}
```

Comment:
This test case and the others with 50ms blocking timeout seem to have the potential to be flaky, and will run very slowly. I will think about other ways to test this behavior - @pawels-optimizely @mikeng13 @mikecdavis any ideas?